### PR TITLE
Clarify rapids_cuda_init_architectures behavior

### PR DESCRIPTION
**Pull request by robertmaynard**

### CODE REVIEW

File: rapids-cmake/cuda/init_architectures.cmake [hunk 1]
```diff
@@ -28,16 +28,17 @@ to include support for `ALL` and `NATIVE` to make CUDA architecture compilation
 
     rapids_cuda_init_architectures(<project_name>)
 
-Establishes what CUDA architectures that will be compiled for.
+Used before enabling the CUDA language either via :cmake:command:`project() <cmake:command:project>` or
+:cmake:command:`enable_language() <cmake:command:enable_language>` to establish te CUDA architectures to be compiled for.
```
Comment (bdice) [inline rapids-cmake/cuda/init_architectures.cmake:32]:
```suggestion
:cmake:command:`enable_language() <cmake:command:enable_language>` to establish the CUDA architectures to be compiled for.
```

File: rapids-cmake/cuda/init_architectures.cmake [hunk 2]
```diff
@@ -28,16 +28,17 @@ to include support for `ALL` and `NATIVE` to make CUDA architecture compilation
...
 Parses the :cmake:variable:`CMAKE_CUDA_ARCHITECTURES <cmake:variable:CMAKE_CUDA_ARCHITECTURES>` for special
 values `ALL`, `NATIVE` and `""`.
 
 .. note::
-  Required to be called before :cmake:command:`project() <cmake:command:project>`
+  Required to be called before the first :cmake:command:`project() <cmake:command:project>` call
```
Comment (bdice) [inline rapids-cmake/cuda/init_architectures.cmake:37]: All the notes I could see use complete sentences, so add a period here.
```suggestion
  Required to be called before the first :cmake:command:`project() <cmake:command:project>` call.
```

**Review by bdice — APPROVED** [review]
Good clarification. I have a couple minor suggestions.

**Comment by robertmaynard** [comment]
> Good clarification. I have a couple minor suggestions.

Thanks for the grammar corrections, applied.

**Comment by robertmaynard** [comment]
@gpucibot merge

**Pull request merged**